test/lib/bamboo/mandrill_adapter_test.exs

```diff
@@ -0,0 +1,98 @@
+defmodule Bamboo.MandrillAdapterTest do
+  use ExUnit.Case
+
+  alias Bamboo.Email
+  alias Bamboo.MandrillAdapter
+
+  @api_key "123_abc"
+
+  Application.put_env(:bamboo, __MODULE__.Mailer, adapter: MandrillAdapter, api_key: @api_key)
+
+
+  defmodule Mailer do
+    use Bamboo.Mailer, otp_app: :bamboo
+  end
+
+  defmodule FakeMandrill do
+    use Plug.Router
+
+    plug Plug.Parsers,
+      parsers: [:urlencoded, :multipart, :json],
+      pass: ["*/*"],
+      json_decoder: Poison
+    plug :match
+    plug :dispatch
+
+    def start_server(parent) do
+      Agent.start_link(fn -> HashDict.new end, name: __MODULE__)
+      Agent.update(__MODULE__, &HashDict.put(&1, :parent, parent))
+      Plug.Adapters.Cowboy.http __MODULE__, [], port: 4001
+    end
+
+    post "/api/1.0/messages/send.json" do
+      conn |> send_resp(200, "SENT") |> send_to_parent
+    end
+
+    defp send_to_parent(conn) do
+      parent = Agent.get(__MODULE__, fn(set) -> HashDict.get(set, :parent) end)
+      send parent, {:fake_mandrill, conn}
+      conn
+    end
+  end
```

Comment:
I'm going to work on extracting this to it's own module in `test/support` next. Should make this a lot simpler and easier to read, also it would be reusable for other adapters in the future